cluster-operator/src/main/java/io/strimzi/operator/cluster/model/ClusterOperatorPKCS12AuthIdentity.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.operator.cluster.model;
+
+import io.fabric8.kubernetes.api.model.Secret;
+import io.strimzi.operator.common.Util;
+
+/**
+ * Represents the identity used by the cluster operator during TLS client authentication in the PKCS12 format.
+ * Can be used by clients that are unable to use the PEM format provided by PemAuthIdentity.
+ */
+public class ClusterOperatorPKCS12AuthIdentity {
```

Comment:
This is currently only used by the ZooKeeper reconcile loop, all other components are using PEM, which is also what I think we want moving forwards. So I chose to put it in cluster-operator to limit it's scope, since the norm should be to use the PEM version. Then hopefully it could be removed when ZooKeeper is no longer required. Happy to move it if others also think it is weird to place it here but that was my thought process behind it.

Reply:
I think ZooKeeper can use PEM files now. I had it working here: https://github.com/strimzi/strimzi-kafka-operator/compare/main...scholzj:strimzi-kafka-operator:use-pem-files?expand=1#diff-e5e5e71484b286fed9dc8f2d7cf2003356fb2f441d60521cb2da028026bb0a43

Should we first move the ZooKeeper clients to PEM and get rid of this? I can have a look into it @katheris if you want. Or you can do it your self if you want either as a separate PR or as part of this PR.

Reply:
Thanks @scholzj  that's useful to know. There's a follow up PR I want to do to align the way we create truststores and keystores as we do it a few different ways in different places, shall I include it in that change? Since the migration code was added there are now a few places we construct a ZK client and the code is duplicated currently I believe

Reply:
Would it make sense to do that first to have this PR cleaner?

Reply:
It would make this PR cleaner, but I'm hesitant do it first because this PR touches quite a few commonly changed classes so I've had to rebase it multiple times already, so it would be good to get it in sooner than later. Plus the changes in this PR will make the truststore/keystore refactor simpler to do and review, since in this PR I made some small changes to close the gap a little. For example in `KafkaAgentClient` using the `X509Certificate` class directly, rather than the `Certificate` class.

Reply:
Ok, than lets do it later.

Reply:
I was also wondering why it's not called just `PKCS12AuthIdentity`. Isn't it the PKCS12 counterpart of the `PemAuthIdentity`? Isn't it good for better abstraction removing that `ClusterOperator` prefix from the class name? I see this class hosts `SECRET_KEY` which is specific to the cluster operator but maybe it should be something in a corresponding derived class? So I mean a PKCS12 auth identity class specific for cluster operator (even if we don't have any more in the end). I see something similar in the `PemAuthIdentity` which makes that abstraction by getting a `secretCertName` parameter to know which key to access in the hosted Secret, or?